### PR TITLE
GO-2140 Event objectClose on logout and object deletion

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-2140/events-for-closed-subscriptions-and-documents

Send objectClose event on deletion and logout

### CODE REVIEW

File: core/block/delete.go [hunk 1]
```diff
@@ -78,6 +78,7 @@ func (s *Service) DeleteObjectByFullID(id domain.FullID) (err error) {
 	}
 
 	sendOnRemoveEvent(s.eventSender, id.ObjectID)
+	sendOnCloseEvent(s.eventSender, id.ObjectID)
```
Comment (fat-fellow) [inline core/block/delete.go:81]: can we test it?

Comment (KirillSto) [reply core/block/delete.go]: Yeah, I have added unit-test on event broadcasting and also moved deletion logics of derived objects to separate func

File: core/block/delete.go [hunk 2]
```diff
@@ -33,33 +34,11 @@ func (s *Service) DeleteObjectByFullID(id domain.FullID) (err error) {
...
-		if err != nil {
-			return fmt.Errorf("on delete: %w", err)
-		}
-		if sbType == coresb.SmartBlockTypeRelation {
-			err := s.deleteRelationOptions(relationKey)
-			if err != nil {
-				return fmt.Errorf("failed to delete relation options of deleted relation: %w", err)
-			}
-		}
+	case coresb.SmartBlockTypeObjectType, coresb.SmartBlockTypeRelation,
```
Comment (fat-fellow) [inline core/block/delete.go:39]: align in a column for readability

**Review by fat-fellow — APPROVED** [review]

File: core/block/delete.go (hunk 2 above)
Comment (KirillSto) [reply core/block/delete.go]: Done!

**Review by cheggaaa — CHANGES_REQUESTED** [review]
We should make it more reliable. 
Good candidates to use:
```
smartblock.Close()
smartblock.ObjectClose()
smartblock.ObjectCloseAllSessions()
```

**Review by cheggaaa — APPROVED** [review]

**Pull request merged**